### PR TITLE
Add wrapper and test-data for picard_CollectHsMetrics

### PR DESCRIPTION
**Pull request by puva**
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [x] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

File: tools/picard/picard_CollectHsMetrics.xml [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+    picard
+    CollectHsMetrics
+    INPUT='$escaped_element_identifier'
+    OUTPUT="${outFile}"
+    MINIMUM_MAPPING_QUALITY="${minimum_mapping_quality}"
+    MINIMUM_BASE_QUALITY="${minimum_base_quality}"
+    COVERAGE_CAP="${coverage_cap}"
+    CLIP_OVERLAPPING_READS="${clip_overlapping_reads}"
+    BAIT_INTERVALS="${baitFile}"
```
Comment (nsoranzo) [inline tools/picard/picard_CollectHsMetrics.xml]:
```suggestion
    BAIT_INTERVALS='${baitFile}'
```

File: tools/picard/picard_CollectHsMetrics.xml [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    TARGET_INTERVALS="${targetFile}"
```
Comment (nsoranzo) [inline tools/picard/picard_CollectHsMetrics.xml]:
```suggestion
    TARGET_INTERVALS='${targetFile}'
```

File: tools/picard/picard_CollectHsMetrics.xml [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+      <param name="coverage_cap" value="200" />
+      <param name="clip_overlapping_reads" value="true	" />
+      <param name="baitFile" value="picard_CollectHsMetrics.baits" ftype="picard_interval_list" />
+      <param name="targetFile" value="picard_CollectHsMetrics.target" ftype="picard_interval_list" /> 
+      <param name="inputFile" value="picard_CollectHsMetrics.bam" ftype="bam" />
+      <output name="outFile" file="picard_CollectHsMetrics_test1.tab" ftype="tabular" lines_diff="6"/>
+    </test>
+  </tests>
+
+  <help>
```
Comment (nsoranzo) [inline tools/picard/picard_CollectHsMetrics.xml]:
```suggestion
  <help><![CDATA[
```

File: tools/picard/picard_CollectHsMetrics.xml [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+  COVERAGE_CAP=Integer
+  CAP=Integer                   Treat bases with coverage exceeding this value as if they had coverage at this value.
+                                Default value: 200.
+  CLIP_OVERLAPPING_READS=Boolean
+                                If true, clip overlapping reads, false otherwise. Default value: true.
+
+@more_info@
+
+  </help>
```
Comment (nsoranzo) [inline tools/picard/picard_CollectHsMetrics.xml]:
```suggestion
  ]]></help>
```

File: tools/picard/picard_CollectHsMetrics.xml [hunk 5]
```diff
@@ -0,0 +1,100 @@
+<tool name="CollectHsMetrics" id="picard_CollectHsMetrics" version="@TOOL_VERSION@.@WRAPPER_VERSION@">
+  <description>compute metrics about datasets generated through hybrid-selection (e.g. exome)</description>
```
Comment (nsoranzo) [inline tools/picard/picard_CollectHsMetrics.xml]: Can you please use 4 spaces for indentation in this file?

File: tools/picard/picard_CollectHsMetrics.xml [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+  <outputs>
+    <data format="tabular" name="outFile" label="${tool.name} on ${on_string}: Summary data"/>
+  </outputs>
+
+  <tests>
+    <test>
+      <param name="minimum_mapping_quality" value="20" />
+      <param name="minimum_base_quality" value="20" />
+      <param name="coverage_cap" value="200" />
+      <param name="clip_overlapping_reads" value="true	" />
```
Comment (nsoranzo) [inline tools/picard/picard_CollectHsMetrics.xml]:
```suggestion
      <param name="clip_overlapping_reads" value="true" />
```

File: tools/picard/picard_CollectHsMetrics.xml (hunk 5 above)
Comment (puva) [reply tools/picard/picard_CollectHsMetrics.xml]: Fixed in https://github.com/galaxyproject/tools-iuc/pull/2415/commits/c31613725a3ac46465ddcb5c7df0794c1435b218

File: tools/picard/picard_CollectHsMetrics.xml [hunk 7]
```diff
@@ -0,0 +1,100 @@
+<tool name="CollectHsMetrics" id="picard_CollectHsMetrics" version="@TOOL_VERSION@.@WRAPPER_VERSION@">
+  <description>compute metrics about datasets generated through hybrid-selection (e.g. exome)</description>
+  <macros>
+    <import>picard_macros.xml</import>
+    <token name="@WRAPPER_VERSION@">1</token>
```
[the remark this answers is not in the document]
Comment (puva) [reply tools/picard/picard_CollectHsMetrics.xml]: Fixed in https://github.com/galaxyproject/tools-iuc/pull/2415/commits/c31613725a3ac46465ddcb5c7df0794c1435b218

File: tools/picard/picard_CollectHsMetrics.xml [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+        <param name="targetFile" type="data" format="picard_interval_list" label="An interval list file that contains the locations of the targets" help="If empty, upload or convert from BED" />
+        <param name="minimum_mapping_quality" type="integer" value="20" label="Minimum mapping quality for a read to contribute coverage" help="MINIMUM_MAPPING_QUALITY; default=20" />
+        <param name="minimum_base_quality" type="integer" value="20" label="Minimum base quality for a base to contribute coverage" help="MINIMUM_BASE_QUALITY; default=20" />
+        <param name="coverage_cap" type="integer" value="200" label="Treat bases with coverage exceeding this value as if they had coverage at this value" help="COVERAGE_CAP; default=200" />
+        <param name="clip_overlapping_reads" type="boolean" label="Clip overlapping reads" checked="true" truevalue="true" falsevalue="false" help="CLIP_OVERLAPPING_READS; default=true" />
+        <expand macro="VS" />
+    </inputs>
+    <outputs>
+        <data name="outFile" format="tabular" label="${tool.name} on ${on_string}: Summary data" />
```
Comment (nsoranzo) [inline tools/picard/picard_CollectHsMetrics.xml]:
```suggestion
        <data name="outFile" format="txt" label="${tool.name} on ${on_string}: Summary data" />
```

By looking at the test data picard_CollectHsMetrics_test1.tab , it doesn't look like a proper tabular file.

File: tools/picard/picard_CollectHsMetrics.xml [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+    <tests>
+        <test>
+            <param name="inputFile" value="picard_CollectHsMetrics.bam" ftype="bam" />
+            <param name="baitFile" value="picard_CollectHsMetrics.baits" ftype="picard_interval_list" />
+            <param name="targetFile" value="picard_CollectHsMetrics.target" ftype="picard_interval_list" />
+            <param name="minimum_mapping_quality" value="20" />
+            <param name="minimum_base_quality" value="20" />
+            <param name="coverage_cap" value="200" />
+            <param name="clip_overlapping_reads" value="true" />
+            <output name="outFile" file="picard_CollectHsMetrics_test1.tab" ftype="tabular" lines_diff="6" />
```
Comment (nsoranzo) [inline tools/picard/picard_CollectHsMetrics.xml]:
```suggestion
            <output name="outFile" file="picard_CollectHsMetrics_test1.tab" lines_diff="6" />
```

File: tools/picard/test-data/picard_CollectHsMetrics_test1.tab [hunk 10]
```diff
@@ -0,0 +1,213 @@
+## htsjdk.samtools.metrics.StringHeader
+# CollectHsMetrics BAIT_INTERVALS=[/tmp/tmpQWQ7TC/files/000/dataset_2.dat] TARGET_INTERVALS=[/tmp/tmpQWQ7TC/files/000/dataset_3.dat] INPUT=picard_CollectHsMetrics_bam OUTPUT=/tmp/tmpQWQ7TC/files/000/dataset_4.dat MINIMUM_MAPPING_QUALITY=20 MINIMUM_BASE_QUALITY=20 CLIP_OVERLAPPING_READS=false COVERAGE_CAP=200 TMP_DIR=[/tmp] VERBOSITY=ERROR QUIET=true VALIDATION_STRINGENCY=LENIENT    METRIC_ACCUMULATION_LEVEL=[ALL_READS] NEAR_DISTANCE=250 SAMPLE_SIZE=10000 COMPRESSION_LEVEL=5 MAX_RECORDS_IN_RAM=500000 CREATE_INDEX=false CREATE_MD5_FILE=false GA4GH_CLIENT_SECRETS=client_secrets.json USE_JDK_DEFLATER=false USE_JDK_INFLATER=false
```
Comment (nsoranzo) [inline tools/picard/test-data/picard_CollectHsMetrics_test1.tab]: Note that `CLIP_OVERLAPPING_READS=false` appears above, while the test specifies `true`. I think this was due to the quoting at https://github.com/galaxyproject/tools-iuc/pull/2415/files#diff-2d8785cfed2944124285a34d2bb46afdR19 , which I have removed.
As a consequence, the test now fails: https://travis-ci.org/galaxyproject/tools-iuc/jobs/539322102
If you agree with my analysis, please update this output file.

File: tools/picard/picard_CollectHsMetrics.xml (hunk 8 above)
Comment (puva) [reply tools/picard/picard_CollectHsMetrics.xml]: Correct!

File: tools/picard/test-data/picard_CollectHsMetrics_test1.tab (hunk 10 above)
Comment (puva) [reply tools/picard/test-data/picard_CollectHsMetrics_test1.tab]: Grazie, good point! I updated the output file. Maybe the error arises from the combination of the removed quotes and the fix at https://github.com/galaxyproject/tools-iuc/pull/2415#discussion_r288951828

**Comment by puva** [comment]
Grazie @nsoranzo !!

**Pull request merged**